akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/EffectfulActorContext.scala

```diff
@@ -91,9 +91,15 @@ import scala.reflect.ClassTag
     override def startTimerWithFixedDelay(key: Any, msg: T, delay: FiniteDuration): Unit =
       startTimer(key, msg, delay, Effect.TimerScheduled.FixedDelayMode)
 
+    override def startTimerWithFixedDelay(key: Any, msg: T, initialDelay: FiniteDuration, delay: FiniteDuration): Unit =
+      startTimer(key, msg, delay, Effect.TimerScheduled.FixedDelayModeWithInitialDelay(initialDelay))
+
```

Comment:
I find it odd to overload a method and add the new param in between existing params. Optional params are better at the end. 

It also increases the chance that someone wants to use the extended method by passing one more param and end up swapping the order. 

if we have
 
`startTimerWithFixedDelay("key", Foo, 5.seconds)` 

and then we update to pass a new param, people will tend to do that as following

`startTimerWithFixedDelay("key", Foo, 5.seconds, 1.second)` 

because it's common practice to add optional params at the end
But now we have initialDelay = 5.seconds and delay = 1.second.

Note that `startTime` is already taking the optional `TimerMode` as last param.

On the other hand, it's also natural to think: I first add the initial delay and then the delay. So, I understand why we are adding in the middle. 

Moreover, the classic scheduler puts the `initialDelay` before the `delay`. In that sense, is good to follow that same pattern.

Reply:
I've chosen to follow the classic scheduler since many are already familiar with that API, and it has a natural feel to it adding the initial delay first like you've said.

But I agree, it has to be decided carefully since it will be hard to change it later.

Reply:
I think following the existing classic API is the right thing to do here (even though I agree with @renato about the general approach).

Reply:
yes, I think I agree to disagree with myself. ;-) 

Let's merge it.

Reply:
👍🏻

Reply:
👍